### ISSUE TITLE
[7.0] Error on files passed through Babel with no plugins/presets and not ignored?

### ISSUE THREAD

**Issue by loganfsmyth** [babel/babel#5454]
Curious for thoughts from people on this. This is something that was brought up back when 6.x was released, since so many people didn't realize you needed to enable plugins, but was tough to fix without being a breaking change, so now seems like a good time to discuss it.

What would people think about throwing an error if you pass a file to Babel but have configured no presets/plugins for it, and have not explicitly ignored the file? We'd of course have to finally fix `ignore` and `only` support so that they actually work as intended.

I think I'd be in favor of adding this, but curious to hear if anyone has concerns.

**Comment by xtuc** [comment]
I have two answer to that:

* It might be due to bad configuration. I believe a lot of error will come from #5451.
Throwing an error would actually help people. If we are more restrictive in the configuration, users would have less problem with it (this doesn't mean less issues here :smile: ).

* Configuration can be passed programmaticly (babelrc.js, Webpack, Starter kit,...). For example an empty list of plugins can be provided because no conditions matched.
Might also due to a bug in user's code, in that case refer to my first point.

**Comment by loganfsmyth** [comment]
> For example an empty list of plugins can be provided because no conditions matched.

Agreed, and that's what stopped us from adding it in 6.x after release.

Maybe we can differentiate an empty array vs no config? That way it's easy to put an empty array if you really want nothing, but people running with no config at all would get a warning? Otherwise we'd have to rely on people to use `ignore` to skip the ones with no plugins, but that does sound not great.
